internal/imagedefinition/README.rst

```diff
@@ -0,0 +1,449 @@
+         # The mirror for apt sources.
+         # Defaults to "http://archive.ubuntu.com/ubuntu/".
+         mirror: <string> (optional)
+         # Ubuntu offers several pockets, which often imply the
+         # inclusion of other pockets. The release pocket only
+         # includes itself. The security pocket includes itself
+         # and the release pocket. Updates includes updates,
+         # security, and release. Proposed includes all pockets.
+         # Defaults to "release".
+         pocket: release | security | updates | proposed (optional)
+         # Used for building an image from a set of archive tasks
+         # rather than seeds. Not yet supported.
+         archive-tasks: (exactly 1 of archive-tasks, seed or tarball must be specified)
+           - <string>
+           - <string>
+         # The seed to germinate from to create a list of packages
+         # to be installed in the image.
+         seed: (exactly 1 of archive-tasks, seed or tarball must be specified)
+             # A list of git, bzr, or http locations from which to
+             # retrieve the seeds.
+             urls: (required if seed dict is specified)
+               - <string>
+               - <string>
+             # The names of seeds to use from the germinate output.
+             # Examples: server, minimal, cloud-image.
+             names: (required if seed dict is specified)
+               - <string>
+               - <string>
+             # Whether to use the --vcs flag when running germinate.
+             # Defaults to "true".
+             vcs: <boolean> (optional)
+             # An alternative branch to use while retrieving seeds
+             # from a git or bzr source.
+             branch: <string> (optional)
+         # Used for pre-built root filesystems rather than germinating
+         # from a seed or using a list of archive-tasks.
+         tarball: (exactly 1 of archive-tasks, seed or tarball must be specified)
+             # The path to the tarball. Can be a local path or an URL.
+             url: <string> (required if tarball dict is specified)
+             # URL to the gpg signature to verify the tarball against.
+             gpg: <string> (optional)
+             # SHA256 sum of the tarball used to verify it has not
+             # been altered.
+             sha256sum: <string> (optional)
+       # ubuntu-image supports building automatically with some
+       # customizations to the image. Note that if customization
+       # is specified, at least one of the subkeys should be used
+       customization: (optional)
+         # Used only for installer images
+         installer: (optional)
+           preseeds: (optional)
+             - <string>
+             - <string>
+           # Only applicable to subiquity based layered images.
+           layers: (optional)
+             - <string>
+             - <string>
+         # Used to create a custom cloud-init configuration.
+         cloud-init: (optional)
```

Comment:
IIRC the cloud-init yaml data that is being submitted should be submitted as a string (in yaml format). Should we make sure that's mentioned here? Like, "<yaml as string>" instead of just "<yaml>"?